linera-execution/src/evm/revm.rs

```diff
@@ -1001,16 +773,64 @@ impl<Runtime: ContractRuntime> CallInterceptorContract<Runtime> {
             // The EVM smart contract is being called
             return Ok(None);
         }
+        // Handling the balances.
+        if let CallValue::Transfer(value) = inputs.value {
+            let source: AccountOwner = inputs.caller.into();
+            let owner: AccountOwner = inputs.bytecode_address.into();
+            if value != U256::ZERO {
+                // In Linera, only non-zero transfers matter
+                let mut runtime = context
+                    .db()
+                    .0
+                    .runtime
+                    .lock()
+                    .expect("The lock should be possible");
+                let amount = Amount::try_from(value).map_err(EvmExecutionError::from)?;
+                let chain_id = runtime.chain_id()?;
+                let destination = Account { chain_id, owner };
+                runtime.transfer(source, destination, amount)?;
+            }
+            // In Ethereum, all transfers matter
+            if let Some(i) =
```

Comment:
Let's call it `err` or `error` instead of `i`.

Reply:
Ok (it came from the REVM code).